extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java

```diff
@@ -917,14 +919,34 @@ public Builder setTableDefinition(final TableDefinition tableDefinition) {
             return this;
         }
 
+        private void initIndexColumns() {
+            if (indexColumns.isEmpty()) {
+                indexColumns = Optional.of(new ArrayList<>());
+            }
+        }
+
         /**
-         * Arrays containing the column names for indexes to persist. The write operation will store the index info as
+         * Add a list of columns to persist together as indexes. The write operation will store the index info as
          * sidecar tables. This argument is used to narrow the set of indexes to write, or to be explicit about the
          * expected set of indexes present on all sources. Indexes that are specified but missing will be computed on
          * demand.
          */
-        public Builder setIndexColumns(final String[][] indexColumns) {
-            this.indexColumns = Optional.ofNullable(indexColumns);
+        public Builder addIndexColumns(final String... indexColumns) {
+            initIndexColumns();
+            this.indexColumns.get().add(indexColumns);
+            return this;
+        }
+
+        /**
+         * Add a collection of index columns to persist. The write operation will store the index info as sidecar
+         * tables. This argument is used to narrow the set of indexes to write, or to be explicit about the expected set
+         * of indexes present on all sources. Indexes that are specified but missing will be computed on demand. To
+         * prevent the generation of index files, use an empty collection.
+         */
+        public Builder setIndexColumns(final Iterable<String[]> indexColumns) {
+            initIndexColumns();
+            final Collection<String[]> indexColumnsCollection = this.indexColumns.get();
+            indexColumns.forEach(indexColumnsCollection::add);
```

Comment:
Shouldn't `set` be replacing the existing? Or assuming empty? Probably a convention question for @devinrsmith .
```suggestion
            final Collection<String[]> indexColumnsCollection = new ArrayList<>;
            indexColumns.forEach(indexColumnsCollection::add);
            this.indexColumns = indexColumnsCollection;
```

Reply:
With immutables library, we typically use the strict builder pattern, which typically prefers "add" and "addAll" style methods. I believe when using non-strict, there may be "set" style methods that error out if it's already set, but I would have to look. Regardless, this method name does seem at odds with the impl.